[New Product] Add RDS MySQL (#2626)
captn3m0/endoflife.date: products/amazon-rds-mysql.md

---
title: Amazon RDS for MySQL

> category: db

Reviewer: ```suggestion
category: service
```

RDS is not a installable database, it is a cloud-service (similar to how we track EKS as service, and Kubernetes as `server-app`).

> iconSlug: mysql

Reviewer: I'd have preferred to use the Amazon RDS Logo, but [the variant simpleicons](https://simpleicons.org/?q=amazonrds) has is unidentifiable (it's meant to be used in architecture diagrams).

The MySQL logo is misleading imo, and the Amazon AWS logo instead fits better.

```suggestion
iconSlug: amazonaws
```

# Auto updates are possible using a custom script.
# Releases and their date are documented on https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/MySQL.Concepts.VersionMgmt.html

releases:
-   releaseCycle: "8.0"
    releaseDate: 2018-10-23
    eol: 2026-04-01
    latest: "8.0.32"
    latestReleaseDate: 2023-02-07

-   releaseCycle: "5.7"
    releaseDate: 2016-02-22
    eol: 2023-10-01
    latest: "5.7.41"
    latestReleaseDate: 2023-02-07

-   releaseCycle: "5.6"
    releaseDate: 2013-07-01
    eol: 2022-03-01
    latest: "unknown"

---

> [Amazon RDS for MySQL](https://aws.amazon.com/rds/mysql) is a PaaS offering from Amazon for
> creating managed MySQL Community Edition databases. RDS makes it easier to set up, operate, and
> scale MySQL deployments on AWS cloud.

Version numbers on Amazon RDS for MySQL are identical to those of [MySQL](/mysql). As a general
guidance, new versions of the MySQL engine become available on Amazon RDS within 5 months of their
general availability by Oracle.

Major versions (`x.y` in Amazon RDS terminology) are supported at least [until the MySQL Community

> Edition end of life](/mysql) and for at least 3 years after their release date on Amazon RDS.

Reviewer: "supported at least until X and for Y" is confusing wording, since support "A&B" is not common usage (does one take precedence, higher/lower etc). Suggest clarifying it with a "minimum" wording, something like this:

```suggestion
Major versions (`x.y` in Amazon RDS terminology) are supported [until the MySQL Community
Edition end of life](/mysql), with a minimum of 3 years from their release date on Amazon RDS.
```

Reviewer 2: Way better indeed.

---
title: Amazon RDS for MySQL
category: db
iconSlug: mysql
permalink: /rds-mysql
releasePolicyLink: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/MySQL.Concepts.VersionMgmt.html

releases:
-   releaseCycle: "8.0"
    releaseDate: 2018-10-23
    eol: 2026-04-01
    latest: "8.0.32"
    latestReleaseDate: 2023-02-07

-   releaseCycle: "5.7"
    releaseDate: 2016-02-22
    eol: 2023-10-01
    latest: "5.7.41"
    latestReleaseDate: 2023-02-07

-   releaseCycle: "5.6"
    releaseDate: 2013-07-01
    eol: 2022-03-01

>     latest: "unknown"

Reviewer 2: Note for myself : check if the `unknown` value (I have added) can be problematic.

Author: At least in the preview it is giving it a crossed out annotation. So there appears to be some kind of handling for the word "unknown"

Reviewer 2: It does not seem to be problematic for now. Let's use it and change it if it becomes problematic.

---

> [Amazon RDS for MySQL](https://aws.amazon.com/rds/mysql) is a PaaS offering from Amazon for
> creating managed MySQL Community Edition databases. RDS makes it easier to set up, operate, and
> scale MySQL deployments on AWS cloud.

> **MySQL recommends that all users run the latest available minor release for whatever major version

Reviewer 2: Is it a MySQL recommendation or an AWS recommandation ?

Author: I looked into this more and MySQL actually doesn't recommend always upgrading to latest because there is no simple reversal.

RDS is managed MySQL. So, while it does makes things easy by doing a backup of the database and makes sure it's stable before applying upgrades. There are 2 major risks involved.
- upgrades can go past maintenance windows and cause unexpected downtime
- upgrades themselves cause downtime (even with Multi-availability zone features enabled) and without proper setup the most stable thing for most production cases is to disable automatic minor upgrades.

Some of this stuff is necessary for my job to know but I agree with your comment. This seems outside the scope of endoflife to comment on this.

I will simply state there are risks and that people should research them on their own.

is in use.**

> - AWS will provide support for major releases 3 years after their RDS release date.

Reviewer 2: Overall that's a good start, but the overall tone must be more neutral. Will try to rephrase those information when I have some time.

Reviewer 2: Some elements are available on https://github.com/endoflife-date/endoflife.date/wiki/Style-Guide.

Author: Awesome I've rewritten my markdown to be more in line with this guide. Some of it could maybe be written more neutral but I'll leave that for review.

Reviewer 2: Ok, thank you @CodaBool.

Reviewer 2: I wrote the following description based on the information you and amazon documentation provided. Here is what it looks like :

```
Version numbers on Amazon RDS for MySQL are identical to those of [MySQL](/mysql). As a general
guidance, new versions of the MySQL engine become available on Amazon RDS within 5 months of their
general availability by Oracle.

Major versions (`x.y` in Amazon RDS terminology) are supported at least [until the MySQL Community
Edition end of life](/mysql) and for at least 3 years after their release date on Amazon RDS.
Minor versions (`x.y.z` in Amazon RDS terminology) are supported at least for 1 year after their release date on Amazon
RDS. Note that in some cases Amazon may deprecate specific major or minor versions sooner, such as
when there are security issues.

Depending on the configuration, the kind of version and their deprecation status, [upgrades can be
manual, automatic or forced](https://aws.amazon.com/rds/faqs/#How_do_I_control_if_and_when_the_engine_version_of_my_DB_instance_is_upgraded_to_new_supported_versions.3F).
When a minor release is deprecated, users are expected to upgrade within a 3 months period. This
period is increased to 6 months for major releases.

Upgrades are performed during the configured scheduled maintenance windows. These windows are
initially set automatically by AWS but can be overridden in the AWS console.

For the most up-to-date information about the Amazon RDS deprecation policy for MySQL, see [Amazon
RDS FAQs](http://aws.amazon.com/rds/faqs/).
```

I completely removed recommandations and focused the description on RDS versions and their support. How do you find it ? Is there error in this description (I do not know RDS).

Author: That looks great. All this info is correct.

Reviewer 2: Ok, cool. I updated your PR. If you intend to modify this PR do not forget to update you branch before ;).

- AWS will provide support for minor versions 1 year after their RDS release date.

Keep in mind that by default minor versions are automatically upgraded during maintenance windows.

> For more info on how RDS versions are deprecated see the AWS [documentation](https://aws.amazon.com/rds/faqs/#What_happens_when_an_Amazon_RDS_DB_engine_version_is_deprecated.3F).

Reviewer 2: The content of this link should be included (and if possible resumed) in the description. The link must be kept though.

Author: I've written out most of the important details now from this link.

> Please follow [best practices](https://aws.amazon.com/blogs/database/best-practices-for-upgrading-amazon-rds-for-mysql-and-amazon-rds-for-mariadb) when performing upgrades to your RDS instance.

Reviewer 2: I do not think that's necessary : IMHO the main purpose of endoflife.date is to provide information on the support of a particular product, not on the upgrade best practices.

Author: I've removed this line now. As stated in another comment. I am just more simply stating in the section that there are risks.

Reviewer: I'd be okay having this. We have similar links for other products where upgrades are considered tough/scary:

1. [AKS](https://endoflife.date/azure-kubernetes-service)
2. [NextCloud](https://endoflife.date/nextcloud#upgrades)
3. [GKE](https://endoflife.date/google-kubernetes-engine)

### Minor Version Support

| Release | Security Support | RDS Release |

> | ------- | ---------------- | ----------- |

Reviewer 2: > I don't want to be making a PR every other month to update this minor table. So, I think I will omit this section since it's not even added as an API.

I think you are right.

[Your script](https://github.com/CodaBool/scrape-rds-eol/tree/main) can be integrated on https://github.com/endoflife-date/release-data, and it will generate a JSON file similar to https://github.com/endoflife-date/release-data/blob/main/releases/mysql.json. Unfortunately the format does not accept additional field yet, but that's something that should change : https://github.com/endoflife-date/release-data/issues/51 (we do not have any date to provide yet).

Author: The scripts I see there are all in Python. Would the script need to be in Python?

Author: I have now removed this Minor version table section in latest commit. Seems there is no current fit for this data.

Reviewer 2: > The scripts I see there are all in Python. Would the script need to be in Python?

Nope, we also have Ruby scripts. I think @captn3m0 wanted to make things quite open regarding the language used, but let's ask him.

Reviewer: Yes, I intentionally want to keep scripts simple, and standalone, and easy to write (read a few pages, and generate a few files).

Contributions are welcome in all major languages.
